[BUG] repo: post-install scripts cannot access repo.xml descriptor

**Describe the bug**

The repo.xml file is not stored until after the script defined in the `<finish>` element (conventionally, "post-install.xql"), has been executed. As a result, the post-install.xql file cannot access data in repo.xml, such as its `<permissions>` element.

**Expected behavior**

According to the documentation's explanation of the `<finish>` element:

> `<finish>`: Like `<prepare>`, this element must point to an XQuery script. This will be executed *after* all data has been uploaded to the database.

(Emphasis in the original.)

Thus, the script defined in the `<finish>` element should run *after* all package assets have been installed.

**To Reproduce**

Many apps contain a stock config.xqm module, with a function, `config:repo-descriptor()`, which returns the repo.xml file. Calling this function in a post-install script fails with the error:

> Caused by: org.exist.xquery.XPathException: exerr:ERROR The actual return type does not match the sequence type declared in the function's signature: config:repo-descriptor() element(). Expected cardinality: exactly one, got 0. [at line 58, column 40, source: /Users/joe/Library/Application Support/org.exist/expathrepo/public-repo-2.0.0/post-install.xq]

The source of the problem is clear enough when viewing Deployment.java, which handles package installation:

https://github.com/eXist-db/exist/blob/303ca0db511ee93bac1b50ccede357fde9658e7b/exist-core/src/main/java/org/exist/repo/Deployment.java#L446-L454

I believe the fix should be to move line 454 to before line 446.

**Context (please always complete the following information):**
 - OS: macOS 11.2.1
 - eXist-db version: 5.3.0-SNAPSHOT (current develop)
 - Java Version: Liberica OpenJDK 1.8.0_28

**Additional context**
- How is eXist-db installed? Built from source
- Any custom changes in e.g. `conf.xml`? No

--- a/exist-core/src/main/java/org/exist/repo/Deployment.java
+++ b/exist-core/src/main/java/org/exist/repo/Deployment.java
@@ -362,7 +362,7 @@ public class Deployment {
                 runQuery(broker, null, packageDir, setupPath.get(), pkgName, QueryPurpose.SETUP);
                 return Optional.empty();
             } else {
-                // otherwise copy all child directories to the target collection
+                // otherwise create the target collection
                 XmldbURI targetCollection = null;
                 if (userTarget != null) {
                     try {
@@ -431,18 +431,22 @@ public class Deployment {
                     runQuery(broker, targetCollection, packageDir, preSetupPath.get(), pkgName, QueryPurpose.PREINSTALL);
                 }
 
-                // any required users and group should have been created by the pre-setup query.
-                // check for invalid users now.
+                // create the group specified in the permissions element if needed
+                // create the user specified in the permissions element if needed; assign it to the specified group
+                // TODO: if the user already exists, check and ensure the user is assigned to the specified group
                 if(requestedPerms.isPresent()) {
                     checkUserSettings(broker, requestedPerms.get());
                 }
 
                 final InMemoryNodeSet resources = findElements(repoXML,RESOURCES_ELEMENT);
 
-                // install
+                // store all package contents into database, using the user/group/mode in the permissions element. however:
+                // 1. repo.xml is excluded for now, since it may contain the default user's password in the clear
+                // 2. contents of directories identified in the path attribute of any <resource path=""/> element are stored as binary
                 final List<String> errors = scanDirectory(broker, transaction, packageDir, targetCollection, resources, true, false,
                         requestedPerms);
                 
+                // store repo.xml, filtering out the default user's password
                 storeRepoXML(broker, transaction, repoXML, targetCollection, requestedPerms);
 
                 // run the post-setup query if present

